File: DynamicProgramming/fibMemo.js
/**
 * @param {number} n
 * @param {*} memo
 * @returns {number}
 */
function fibMemo(n, memo = {}) {
  if (memo[n]) return memo[n];
  if (n <= 1) return n;
  
  memo[n] = fibMemo(n - 1, memo) + fibMemo(n-2, memo);
  return memo[n]
}

console.log(fibMemo(7))

File: DynamicProgramming/fibMemo.cpp
#include <iostream>
#include <vector>
using namespace std;

int fibMemo(int n, vector<int> memo = {})
{
    if (memo[n])
        return memo[n];
    if (n <= 1)
        return n;

    memo[n] = fibMemo(n - 1, memo) + fibMemo(n - 2, memo);
    return memo[n];
}
int main()
{
    cout <<"fibonacci of 5: "<< fibMemo(5);
    return 0;
}